### PR TITLE
vx_amd_migraphx tests

### PR DESCRIPTION
**Pull request by LakshmiKumar23**
Updated scripts to take multiple batch sizes. However with TOT MIGraphX repo only batch size = 1 works. TO test on bigger batch sizes , build and install [this branch](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/tree/dyn_reshape_parsing)
Results on confluence page

### CODE REVIEW

File: tests/amd_migraphx_test/mnist/migraphx_node_test.cpp [hunk 1]
```diff
@@ -122,16 +142,18 @@ int main(int argc, char **argv) {
         return status;
     }
 
+    //copy results into file
+    outputFile.open(date + "/mnist-output-results.csv");
+    outputFile << "iteration,Randomly chosen digit,Result from inference,result\n";
```
Comment (asalmanp) [inline tests/amd_migraphx_test/mnist/migraphx_node_test.cpp:147]: add space after each comma.

File: tests/amd_migraphx_test/mnist/migraphx_node_test.cpp [hunk 2]
```diff
@@ -122,16 +142,18 @@ int main(int argc, char **argv) {
...
+
+    float *output_buf = (float*)ptr;
     auto num_results = 10;
-    float* max = std::max_element((float*)ptr, (float*)ptr + num_results);
-    int answer = max - (float*)ptr;
+    for(int i = 0; i < batch_size; i++, output_buf+= num_results) {
```
Comment (asalmanp) [inline tests/amd_migraphx_test/mnist/migraphx_node_test.cpp:151]: add a space before +=

File: tests/amd_migraphx_test/mnist/migraphx_node_test.cpp [hunk 3]
```diff
@@ -122,16 +142,18 @@ int main(int argc, char **argv) {
...
+        float* max = std::max_element(output_buf, output_buf + num_results);
+        int answer = max - output_buf;
 
-    std::cout << std::endl
-              << "Randomly chosen digit: " << rand_digit << std::endl
-              << "Result from inference: " << answer << std::endl
-              << std::endl
-              << (answer == rand_digit ? "CORRECT" : "INCORRECT") << std::endl
-              << std::endl;
+        outputFile << i+1 << "," << rand_digit[i] << "," << answer << "," << (answer == rand_digit[i] ? "Correct":"Incorrect") << "\n";
```
Comment (asalmanp) [inline tests/amd_migraphx_test/mnist/migraphx_node_test.cpp:155]: add a space before and after +

File: tests/amd_migraphx_test/resnet50/migraphx_node_test.cpp [hunk 4]
```diff
@@ -92,65 +109,101 @@ int main(int argc, char **argv) {
...
-    float *buf = (float *)malloc(total*sizeof(float));
-    float *R = buf;
-    float *G = R + rows * cols;
-    float *B = G + rows * cols;
-
-    float mean_vec[3] = {0.485, 0.456, 0.406};
-    float stddev_vec[3] = {0.229, 0.224, 0.225};    
-    float preproc_mul[3] = { 1 / (255 * stddev_vec[0]), 1 / (255 * stddev_vec[1]), 1 / (255 * stddev_vec[2])};
-    float preproc_add[3] = {(mean_vec[0] / stddev_vec[0]), (mean_vec[1] / stddev_vec[1]), (mean_vec[2] / stddev_vec[2])};
+    int count = input_dims[0]*input_dims[1]*input_dims[2]*input_dims[3];
```
Comment (asalmanp) [inline tests/amd_migraphx_test/resnet50/migraphx_node_test.cpp:112]: add space before and after each *

File: tests/amd_migraphx_test/resnet50/migraphx_node_test.cpp [hunk 5]
```diff
@@ -92,65 +109,101 @@ int main(int argc, char **argv) {
...
+        DIR *pDir;
+        pDir = opendir(inputFileName.c_str());
+        if(pDir == NULL) {
+            std::cerr <<"ERROR: Image file directory invalid" << std::endl;
+        }
+        int bs = 0;
+        while((pDirent = readdir(pDir)) && bs < batch_size) {
+            std::string imageFileName;
+            bool inference = false;
+            if ((std::string(pDirent->d_name)).size() > 2)
```
Comment (asalmanp) [inline tests/amd_migraphx_test/resnet50/migraphx_node_test.cpp:132]: add curly bracket after the if and else

File: tests/amd_migraphx_test/resnet50/migraphx_node_test.cpp [hunk 6]
```diff
@@ -172,13 +225,19 @@ int main(int argc, char **argv) {
         return status;
     }
 
+    //copy results into file
+    outputFile.open(date + "/resnet50-output-results.csv");
+    outputFile << "image,classification,probability,label\n";
```
Comment (asalmanp) [inline tests/amd_migraphx_test/resnet50/migraphx_node_test.cpp:230]: add space after each comma

File: tests/amd_migraphx_test/resnet50/migraphx_node_test.cpp [hunk 7]
```diff
@@ -172,13 +225,19 @@ int main(int argc, char **argv) {
...
     auto num_results = 1000;
-    int final_argmax_result = std::distance((float*)ptr, std::max_element((float*)ptr, (float*)ptr + num_results));
-    std::string output_label = labelText[final_argmax_result];
-
-    std::cout << "output index = " << final_argmax_result << "  && output label = " << output_label << std::endl;
-
+    for(int i = 0; i < batch_size; i++, output_buf += num_results) {
+        int final_argmax_result = std::distance(output_buf, std::max_element(output_buf, output_buf + num_results));
+        std::string output_label = labelText[final_argmax_result];
+        outputFile << i+1 << "," << final_argmax_result << "," << output_buf[final_argmax_result] << "," << output_label.c_str() << "\n";
```
Comment (asalmanp) [inline tests/amd_migraphx_test/resnet50/migraphx_node_test.cpp:238]: add space before and after +

File: tests/amd_migraphx_test/runMIGraphXTests/runMIGraphXTests.cpp [hunk 8]
```diff
@@ -235,521 +231,453 @@ int main(int argc, char **argv) {
...
+                ERROR_CHECK_STATUS(vxVerifyGraph(graph_googlenet));
+                ERROR_CHECK_STATUS(vxProcessGraph(graph_googlenet));
+
+                //googlenet timing for 1000 iterations
+                t0 = clockCounter();
+                for(int i = 0; i < N; i++) {
+                    ERROR_CHECK_STATUS(vxProcessGraph(graph_googlenet));
+                }
+                t1 = clockCounter();
+                float googlenetTime = (float)(t1-t0)*1000.0f/(float)freq/(float)N;
```
Comment (asalmanp) [inline tests/amd_migraphx_test/runMIGraphXTests/runMIGraphXTests.cpp:462]: add a space before and after -, and /

**Review by asalmanp — CHANGES_REQUESTED** [review]
@LakshmiKumar23 There are many minor formatting issues with this PR. Please go through all the files and make sure that you follow consistent formatting across all the files. Thanks.

File: tests/amd_migraphx_test/mnist/CMakeLists.txt [hunk 9]
```diff
@@ -31,12 +31,31 @@ set(ROCM_PATH /opt/rocm CACHE PATH "default ROCm installation path")
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
 include_directories(${ROCM_PATH}/include ${ROCM_PATH}/include/mivisionx)
 link_directories(${ROCM_PATH}/lib)
+include_directories(${OpenCV_INCLUDE_DIRS})
```
[the remark this answers is not in the document]
Comment (LakshmiKumar23) [reply tests/amd_migraphx_test/mnist/CMakeLists.txt:34]: @asalmanp Oh let me undo this. It's only need for the resnet50 test. Even the entire test suite doesn't need it

File: tests/amd_migraphx_test/mnist/migraphx_node_test.cpp [hunk 10]
```diff
@@ -57,12 +63,13 @@ void read_input_digit(const int n, std::vector<float>& input_digit) {
...
     
-    if(argc < 2) {
-        std::cout << "Usage: \n ./migraphx_node_test <path-to-resnet50 ONNX model>" << std::endl;
+    if(argc < 4) {
+        std::cout << "Usage: \n ./migraphx_node_test <path-to-mnist ONNX model> --batch-size n" << std::endl;
         return -1;
     }
     
     std::string modelFileName = argv[1];
+    vx_size batch_size = stoul(argv[3]);
```
[the remark this answers is not in the document]
Comment (LakshmiKumar23) [reply tests/amd_migraphx_test/mnist/migraphx_node_test.cpp:72]: With TOT MIGraphX only batch size 1, with the link I posted, I have tried till 128

File: tests/amd_migraphx_test/runMIGraphXTests/runMIGraphXTests.cpp [hunk 11]
```diff
@@ -233,523 +221,453 @@ int main(int argc, char **argv) {
...
+                ERROR_CHECK_STATUS(vxVerifyGraph(graph_resnet50));
+                ERROR_CHECK_STATUS(vxProcessGraph(graph_resnet50));
+
+                //resnet50 timing for 1000 iterations
+                t0 = clockCounter();
+                for(int i = 0; i < N; i++) {
+                    ERROR_CHECK_STATUS(vxProcessGraph(graph_resnet50));
+                }
+                t1 = clockCounter();
+                float resnet50Time = (float)(t1-t0) * 1000.0f / (float)freq / (float)N;
```
Comment (asalmanp) [inline tests/amd_migraphx_test/runMIGraphXTests/runMIGraphXTests.cpp:337]: add a space before and after -. there are multiple occurrences of t1-t0 in this file. please fix the formatting for all of them. Thanks.

**Review by asalmanp — CHANGES_REQUESTED** [review]

**Review by asalmanp — APPROVED** [review]
@paveltc please run the MIGraphX tests including the new tests introduced in this PR for the MIGRaphX node. Please check with @LakshmiKumar23 for how to run the tests. Thanks.

**Comment by kiritigowda** [comment]
@LakshmiKumar23 is this ready to be merged?

**Comment by LakshmiKumar23** [comment]
> @LakshmiKumar23 is this ready to be merged?

@kiritigowda No, we need Pavel to run tests and approve it

**Comment by kiritigowda** [comment]
@paveltc let me know when this good to merge.

**Comment by paveltc** [comment]
@asalmanp @LakshmiKumar23 @kiritigowda I ran the tests for this PR. It's good to merge.

**Pull request merged**